protected/controllers/survey/AjaxSave.php

```diff
@@ -0,0 +1,53 @@
+<?php
+declare(strict_types=1);
+
+namespace prime\controllers\survey;
+
+use prime\helpers\ModelHydrator;
+use prime\interfaces\AccessCheckInterface;
+use prime\models\ar\Permission;
+use prime\models\ar\Survey;
+use prime\models\survey\SurveyForCreate;
+use prime\repositories\SurveyRepository;
+use prime\values\SurveyId;
+use yii\base\Action;
+use yii\web\BadRequestHttpException;
+use yii\web\Request;
+use yii\web\Response;
+
+class AjaxSave extends Action
+{
+    public function run(
+        AccessCheckInterface $accessCheck,
+        ModelHydrator $hydrator,
+        Request $request,
+        SurveyRepository $repository,
+        Response $response,
+        null|int $id = null,
+    ) {
+        $response->format = Response::FORMAT_JSON;
+
+        if (is_null($id)) {
```

Comment:
This is a code smell, you have 2 use cases: create and update.
In multiple places in the function you check the same binary option and use different behavior.